examples/webserver/Makefile

```diff
@@ -0,0 +1,251 @@
+	$(CC) -c $(CFLAGS) $(SDDF)/util/putchar_debug.c -o $(BUILD_DIR)/sddf_putchar_debug.o
+
+$(BUILD_DIR)/sddf_putchar_serial.o: $(SDDF)/util/putchar_serial.c
+	$(CC) -c $(CFLAGS) $(SDDF)/util/putchar_serial.c -o $(BUILD_DIR)/sddf_putchar_serial.o
+
+$(BUILD_DIR)/timer.o: $(TIMER_DRIVER)/timer.c
+	$(CC) -c $(CFLAGS) $< -o $@
+
+$(BUILD_DIR)/timer.elf: $(addprefix $(BUILD_DIR)/, $(TIMER_OBJS))
+	$(LD) $(LDFLAGS) $^ $(LIBS) -o $@
+
+$(BUILD_DIR)/micropython.elf: FORCE mpy-cross $(BUILD_DIR)/sddf_util_cache.o $(BUILD_DIR)/libm/libm.a manifest.py
+	echo "base_dir='$(WEBSITE_DIR)'" >$(BUILD_DIR)/config.py
+	cp manifest.py webserver.py $(BUILD_DIR)
+	make -C $(LIONSOS)/components/micropython -j$(nproc) \
+			MICROKIT_SDK=$(MICROKIT_SDK) \
+			MICROKIT_BOARD=$(MICROKIT_BOARD) \
+			MICROKIT_CONFIG=$(MICROKIT_CONFIG) \
+			BUILD=$(abspath $(BUILD_DIR)) \
+			LIBMATH=$(abspath $(BUILD_DIR)/libm) \
+			ETHERNET_CONFIG_INCLUDE=$(abspath $(ETHERNET_CONFIG_INCLUDE)) \
+			FROZEN_MANIFEST=$(abspath $(BUILD_DIR)/manifest.py) \
+			EXEC_MODULE="webserver.py"
+
+ETHERNET=$(SDDF)/drivers/network/meson/
+$(BUILD_DIR)/meson_ethernet.o: FORCE
+	BUILD_DIR=$(abspath $(BUILD_DIR)) MICROKIT_INCLUDE=$(BOARD_DIR)/include ETHERNET_CONFIG_INCLUDE=$(abspath $(ETHERNET_CONFIG_INCLUDE)) make -C $(ETHERNET)
+
+$(BUILD_DIR)/sddf_util_cache.o: $(SDDF)/util/cache.c
+	$(CC) -c $(CFLAGS) $< -o $@
+
+$(BUILD_DIR)/musllibc/lib/libc.a:
+	make -C $(MUSL) \
+		C_COMPILER=aarch64-none-elf-gcc \
+		TOOLPREFIX=aarch64-none-elf- \
+		CONFIG_ARCH_AARCH64=y \
+		STAGE_DIR=$(abspath $(BUILD_DIR)/musllibc) \
+		SOURCE_DIR=.
+
+$(BUILD_DIR)/libnfs/lib/libnfs.a: $(BUILD_DIR)/musllibc/lib/libc.a
+	MUSL=$(abspath $(BUILD_DIR)/musllibc) cmake -S $(LIBNFS) -B $(BUILD_DIR)/libnfs
+	cmake --build $(BUILD_DIR)/libnfs
+
+$(BUILD_DIR)/nfs/nfs.a: FORCE $(BUILD_DIR)/musllibc/lib/libc.a
+	make -C $(NFS) \
+		BUILD_DIR=$(abspath $(BUILD_DIR)/nfs) \
+		MICROKIT_INCLUDE=$(BOARD_DIR)/include \
+		MUSLLIBC_INCLUDE=$(abspath $(BUILD_DIR)/musllibc/include) \
+		LIBNFS_INCLUDE=$(abspath $(LIBNFS)/include) \
+		ETHERNET_CONFIG_INCLUDE=$(abspath $(ETHERNET_CONFIG_INCLUDE))
+
+$(BUILD_DIR)/nfs.elf: $(BUILD_DIR)/nfs/nfs.a $(BUILD_DIR)/libnfs/lib/libnfs.a $(BUILD_DIR)/musllibc/lib/libc.a
+	$(LD) \
+		$(LDFLAGS) \
+		$(BUILD_DIR)/nfs/nfs.a \
+		-L$(BUILD_DIR)/libnfs/lib \
+		-L$(BUILD_DIR)/musllibc/lib \
+		-L$(LIBGCC) \
+		-lgcc \
+		-lc \
+		$(LIBS) \
+		-lnfs \
+		-o $(BUILD_DIR)/nfs.elf
+
+$(BUILD_DIR)/ethernet.elf: $(BUILD_DIR)/meson_ethernet.o $(BUILD_DIR)/sddf_util_printf.o $(BUILD_DIR)/sddf_putchar_debug.o
+	$(LD) $(LDFLAGS) $^ $(LIBS) -o $@
+
+NETWORK_COMPONENTS := $(SDDF)/network/components
+NETWORK_COMPONENTS_INCLUDE := \
+	-I$(SDDF)/include \
+	-I$(BOARD_DIR)/include \
+	-I$(LWIP)/include \
+	-I$(LWIP)/include/ipv4 \
+	-Iethernet_config
+NETWORK_COMPONENT_CFLAGS := \
+	$(NETWORK_COMPONENTS_INCLUDE) \
+	-mstrict-align \
+	-ffreestanding \
+	-g \
+	-O0 \
```

Comment:
Why `-O0`?

Reply:
I think `-g` is removed now but we want to keep that in case we need to objdump the ELFs or something.